hack/ci-deploy.sh

```diff
@@ -71,7 +71,7 @@ else
   kubectl_create -n=local-csi-driver -f="${SO_CSI_DRIVER_PATH}"
 fi
 
-if [[ -n "${SO_SCYLLACLUSTER_STORAGECLASS_NAME}" ]]; then
+if [[ -n "${SO_SCYLLACLUSTER_STORAGECLASS_NAME:-}" ]]; then
   yq e --inplace '.spec.datacenter.racks[0].storage.storageClassName = env(SO_SCYLLACLUSTER_STORAGECLASS_NAME)' "${DEPLOY_DIR}/manager/50_scyllacluster.yaml"
 elif [[ -n "${SO_SCYLLACLUSTER_STORAGECLASS_NAME+x}" ]]; then
```

Comment:
this checks if it's undefined - shouldn't reversing the order fix it?

Reply:
The logic behind this is roughly
```
if [[ -n "${SO_SCYLLACLUSTER_STORAGECLASS_NAME:-}" ]]; then
  echo "set and nonempty, replace storageClassName"
elif [[ -n "${SO_SCYLLACLUSTER_STORAGECLASS_NAME+x}" ]]; then
  echo "set and empty, delete explicit storageClassName (use default)"
else
  echo "not set, do not modify the manifest"
fi
```

if you reverse the order without adding the substitution you still end up with unbound variable error in case it's not set I believe

Reply:
you are correct, I though +x gets passed this